entity-framework/core/dbcontext-configuration/index.md

```diff
@@ -1,162 +1,393 @@
+            : base(contextOptions)
+        {
+        }
+    }
+-->
+[!code-csharp[ApplicationDbContextBase](../../../samples/core/Miscellaneous/ConfiguringDbContext/InheritDbContext/ApplicationDbContext.cs?name=ApplicationDbContextBase)]
+
+This allows multiple concrete subclasses to call this base constructor using their different generic `DbContextOptions<TContext>` instances. For example:
+
+<!--
+    public sealed class ApplicationDbContext1 : ApplicationDbContextBase
+    {
+        public ApplicationDbContext1(DbContextOptions<ApplicationDbContext1> contextOptions)
+            : base(contextOptions)
+        {
+        }
+    }
+
+    public sealed class ApplicationDbContext2 : ApplicationDbContextBase
+    {
+        public ApplicationDbContext2(DbContextOptions<ApplicationDbContext2> contextOptions)
+            : base(contextOptions)
+        {
+        }
+    }
+-->
+[!code-csharp[ApplicationDbContext12](../../../samples/core/Miscellaneous/ConfiguringDbContext/InheritDbContext/ApplicationDbContext.cs?name=ApplicationDbContext12)]
+
+Notice that this is exactly the same pattern as when inheriting from `DbContext` directly. That is, the `DbContext` constructor itself accepts a non-generic `DbContextOptions` for this reason.
+
+A `DbContext` subclass intended to be both instantiated and inherited from should expose both forms of constructor. For example:
+
+<!--
+    public class ApplicationDbContext : DbContext
+    {
+        public ApplicationDbContext(DbContextOptions<ApplicationDbContext> contextOptions)
+            : base(contextOptions)
+        {
+        }
+
+        protected ApplicationDbContext(DbContextOptions contextOptions)
+            : base(contextOptions)
+        {
+        }
+    }
+-->
+[!code-csharp[ApplicationDbContext](../../../samples/core/Miscellaneous/ConfiguringDbContext/InheritDbContext/ApplicationDbContext.cs?name=ApplicationDbContext)]
+
+## Design-time DbContext configuration
 
-```csharp
-using (var context = serviceProvider.GetService<BloggingContext>())
-{
-  // do stuff
-}
+EF Core design-time tools such as those for [EF Core migrations](xref:core/managing-schemas/migrations/index) need to be able to discover and create a working instance of a `DbContext` type in order to gather details about the application's entity types and how they map to a database schema. This process can be automatic as long as the tool can easily create the `DbContext` in such a way that it will be configured similarly to how it would be configured at run-time.
 
-var options = serviceProvider.GetService<DbContextOptions<BloggingContext>>();
-```
+While any pattern that provides the necessary configuration information to the `DbContext` can work at run-time, tools that require using a `DbContext` at design-time can only work with a limited number of patterns. These are covered in more detail in [Design-Time Context Creation](xref:core/cli/dbcontext-creation).
 
 ## Avoiding DbContext threading issues
```

Comment:
Also either rename "Forgetting to await the completion of an asynchronous operation before starting any other operation on the same DbContext" to "Asynchronous operation pitfalls" or reduce the heading level or convert it to an [!IMPORTANT] block.